tutorials/gf/gf.cpp

```diff
@@ -17,55 +35,137 @@ int main() {
+  index_t     m2(N2);
+  itime_t     m3(beta, N3);
+  legendre_t  m4(beta, N4);
+  chebyshev_t m5(beta, N5);
+
+  // Change the shape of the Green's function
+  g.reshape(m1, m2, m3, m4, m5);
+  std::cout<<"Green's function object of the total size "<<g.data().size()<<" has been created"<<std::endl;
+}
 
-  using matsubara = alps::gf::matsubara_positive_mesh;
-  using index     = alps::gf::index_mesh;
-  using itime     = alps::gf::itime_mesh;
-  using legendre  = alps::gf::legendre_mesh;
-  using chebyshev = alps::gf::chebyshev_mesh;
+// Perform I/O routines with Green's functions
+void greens_function_io(alps::gf::greenf<double, itime_t, legendre_t, chebyshev_t> &g_copy,
+                        alps::gf::greenf_view<double, legendre_t, chebyshev_t> &g_view) {
+  // Save Green's functions into the file
+  std::ofstream f1("green.dat");
+  f1<<g_copy;
+  f1.close();
+  f1.open("green_view.dat");
+  f1<<g_view;
 
-  using iw = alps::gf::matsubara_positive_mesh::index_type;
-  using ii = alps::gf::index_mesh::index_type;
-  using it = alps::gf::itime_mesh::index_type;
-  using il = alps::gf::legendre_mesh::index_type;
-  using ic = alps::gf::chebyshev_mesh::index_type;
+  // Save/load Green's functions into hdf5
+  alps::hdf5::archive ar("gf.h5", "w");
 
-  // Define meshes for the Green's function object
-  matsubara m1(beta, N1);
-  index     m2(N2);
-  itime     m3(beta, N3);
-  legendre  m4(beta, N4);
-  chebyshev m5(beta, N5);
-
-  // create simple Green's function object
-  alps::gf::greenf<double, matsubara, index, itime, legendre, chebyshev> g(m1, m2, m3, m4, m5);
-  
-  std::cout<<"Green's function object of the total size "<<g.data().size()<<" has been created"<<std::endl;
+  // save Green's function object into group 'GF'
+  ar["GF"]<<g_copy;
+  // save Green's function view into group 'GF_VIEW'
+  ar["GF_VIEW"]<<g_view;
+
+  // create copy of the view object
+  alps::gf::greenf<double, legendre_t, chebyshev_t> g_view_copy = g_view;
+
+  // change view object
+  for (int i = 0; i < g_view.mesh1().extent(); ++i) {
+    for (int j = 0; j < g_view.mesh2().extent(); ++j) {
+      g_view(il(i), ic(j)) = i*10+j;
+    }
+  }
+
+  // restore view by loading it from hdf5
+  ar["GF_VIEW"] >> g_view;
+
+  // check that the data has been restored
+  if((g_view - g_view_copy).norm() > 1e-19 ) {
+    throw std::logic_error("Something wrong");
+  }
+}
+
+// Perform basic arithmetics on the Green's function object
+void basic_arithmetics(alps::gf::greenf_view<double, legendre_t, chebyshev_t> &g_view) {
+  // create copy of the view object
+  alps::gf::greenf<double, legendre_t, chebyshev_t> g_view_copy = g_view;
+
+  // Multiply by scalar
+  g_view *= 2.0;
+
+  // check the results
+  for (int i = 0; i < g_view.mesh1().extent(); ++i) {
+    for (int j = 0; j < g_view.mesh2().extent(); ++j) {
+      if(std::abs(g_view(il(i), ic(j)) - 2.0*g_view_copy(il(i), ic(j)) ) > 1e-19 ) {
+        throw std::logic_error("Something wrong");
+      }
+    }
+  }
```

Comment:
A matter of taste: the library wants you to use the typed indices in the loop.  Questionable merit of that idea aside, one should maybe think of having the tutorial follow that convention except maybe one example.